core/src/main/java/org/apache/calcite/avatica/AvaticaUtils.java

```diff
@@ -235,10 +232,29 @@ public static Class box(Class clazz) {
       } catch (NoSuchFieldException e) {
         // ignore
       }
+      assert pluginClass.isAssignableFrom(clazz);
```

Comment:
Assertions are not enabled by default. Would you please replace the assertion with a regular if check?

Reply:
+1 for your suggestion. 
I do not want add assert here, just moved the position from up line to here . Agree with you, I do not like use assert to do the logical judgment.  but I see that the project seems to favor to use assert. For example, in the following classes: `avaticaresultset`, `avaticasite`, `avaticaconnection`, `builtinconnectionproperty`, `connectionconfigimpl` etc. 
Do you know any principles for using assert in this project? // I'm new to this project :)

Reply:
My rule of thumb is to use assertions for situations that should never happen, it's more of a "documentation" for me where I state expectations that follow logically but are maybe not evident, all the other cases need checks+exception

Reply:
I see. Thanks for sharing your thoughts on the use of assert @asolimando , Thank you!